pandas/tseries/resample.py

```diff
@@ -673,18 +670,15 @@ def aggregate(self, arg, *args, **kwargs):
     def _get_new_index(self):
         """ return our new index """
         ax = self.ax
-        ax_attrs = ax._get_attributes_dict()
-        ax_attrs['freq'] = self.freq
-        obj = self._selected_obj
 
         if len(ax) == 0:
-            new_index = PeriodIndex(data=[], **ax_attrs)
-            return obj.reindex(new_index)
-
-        start = ax[0].asfreq(self.freq, how=self.convention)
-        end = ax[-1].asfreq(self.freq, how='end')
+            values = []
+        else:
+            start = ax[0].asfreq(self.freq, how=self.convention)
+            end = ax[-1].asfreq(self.freq, how='end')
```

Comment:
Agree in part - you example valid because it's finding the edges of the Periods (and so casts to edges / `DatetimeIndex`): 

```python
In [73]: s.resample('5D', convention='start').asfreq().index
Out[73]: 
DatetimeIndex(['1990-01-01', '1990-01-06', '1990-01-11', '1990-01-16',
               '1990-01-21', '1990-01-26', '1990-01-31'],
              dtype='datetime64[ns]', freq='5D')
```
...but that doesn't mean it's valid when converted back to Periods.

Rather than `resample` doing this through a `convention` kwarg and changing the type, I think that could be a method to get the edges. I realize this is leaking into Interval Index discussion - that's because that's the concept that Periods are attempting to represent.

CC @shoyer